src/router/utils/getHashString.es6.js

```diff
@@ -0,0 +1,9 @@
+'use strict';
+
+export default function (url) {
+	const hashMatch = url.match(/#([^?]*)/);
```

Comment:
spoke to horwitz. We have to strip the ?string out of the hash for clean matching. I'll put back the changes that add a hashQuery property to the context object.